model/goniometer.py

```diff
@@ -82,9 +88,11 @@ def single_axis_goniometer_from_phil(params, reference=None):
 
         """
 
-        # Check the axes parameter
-        if params.goniometer.axes is not None and len(params.goniometer.axes) != 3:
-            raise RuntimeError("Single axis goniometer requires 3 axes parameters")
+        # Check the axis parameter and copy from axes if required
+        if params.goniometer.axis is None:
+            params.goniometer.axis = params.goniometer.axes
+        if params.goniometer.axis is not None and len(params.goniometer.axis) != 3:
```

Comment:
There should be no need for this size check, as the phil parsing should take care of this thanks to the `.type = floats(size=3)`.

Reply:
this was in the case that `axis` was left as `None` and has been copied over from `axes`